mayastor/src/lvm/lvm.rs

```diff
@@ -0,0 +1,285 @@
+use tokio::process::Command;
+use crate::lvm::error::Error;
+
+use rpc::mayastor::{CreatePoolRequest, Pool, PoolType, PoolState};
+
+const PVCREATE_COMMAND: &str = "pvcreate";
+const PVS_COMMAND: &str = "pvs";
+const PVREMOVE_COMMAND: &str = "pvremove";
+const VGCREATE_COMMAND: &str = "vgcreate";
+const VGS_COMMAND: &str = "vgs";
+const VGREMOVE_COMMAND: &str = "vgremove";
+
+pub async fn lookup_lvm_vg(name: &str) -> Option<Pool> {
+    let pools = list_lvm_vg().await.ok()?;
+    match pools.iter().find(|p| p.name == name) {
+        Some(p) => {
+            return Some(Pool{
+                name: p.name.as_str().to_owned(),
+                disks: p.disks.as_slice().to_owned(),
+                capacity: p.capacity,
+                used: p.used,
+                pooltype: PoolType::Lvm.into(),
+                state: PoolState::PoolOnline.into(),
+            });
+        }
+        None => None
+    }
+}
+
+pub async fn import_or_create_lvm_vg(req: CreatePoolRequest) -> Result<Pool, Error> {
+    let name = req.name.as_str();
+    let disks = req.disks.as_slice();
+    let pool = match import_lvm_vg(CreatePoolRequest{
+            name: name.to_string(),
+            disks: disks.to_owned(),
+            pooltype: PoolType::Lvm.into(),
+        }).await {
+        Ok(p) => p,
+        Err(e) => match e {
+            Error::NotFound{..} => create_lvm_vg(CreatePoolRequest{
+                name: name.to_string(),
+                disks: disks.to_owned(),
+                pooltype: PoolType::Lvm.into(),
+            }).await?,
+            _ => return Err(e)
+        }
+    };
+    info!("The pool '{}' has been created on {:?}", name, disks);
+    Ok(pool)
+}
+
+pub async fn import_lvm_vg(req: CreatePoolRequest) -> Result<Pool, Error> {
+    let pools = list_lvm_vg().await?;
+    match pools.iter().find(|p| p.name == req.name) {
+        Some(p) => {
+            if p.disks == req.disks {
+                return Ok(Pool{
+                    name: p.name.as_str().to_owned(),
+                    disks: p.disks.as_slice().to_owned(),
+                    capacity: p.capacity,
+                    used: p.used,
+                    pooltype: PoolType::Lvm.into(),
+                    state: PoolState::PoolOnline.into(),
+                });
+            }
+            Err(Error::ImportErr {err: format!("lvm vg {} disks not match", req.name)})
```

Comment:
Best for readability if you special case the error case:

```suggestion
            if p.disks != req.disks {
                return Err(Error::ImportErr {err: format!("lvm vg {} disks not match", req.name)});
            }

            Ok(Pool{
                    name: p.name.as_str().to_owned(),
                    disks: p.disks.as_slice().to_owned(),
                    capacity: p.capacity,
                    used: p.used,
                    pooltype: PoolType::Lvm.into(),
                    state: PoolState::PoolOnline.into(),
                });
```